pkg/controller/util/endpointslice/endpointset.go

```diff
@@ -30,6 +30,7 @@ type endpointHash string
 type endpointHashObj struct {
 	Addresses []string
 	Hostname  string
+	Name      string
```

Comment:
the description of the tyoe seems to omit `targetRef`on purpose

Reply:
Right, this has being just a hacky solution to make the test pass, not sure it should be the proper one, maybe we have just to use some conditions preference in case of same endpoints where ready has more prefernece than not ready.

Reply:
I think this change is basically saying "the code was wrong to not consider targetRef before", so the answer is that the comment needs to be updated.

That said, you should probably compare the whole `targetRef`? Or at least both Name and Namespace.

Reply:
I think endpointslices will have endpoints with target ref at the same namespaces so Name should be enough, but maybe we can have the whole targetRef if there is no mayor performance penalty.

Reply:
If we're hashing something, could we also consider the `metadata.uid`?

Reply:
> Yeah on further thought namespace + name would probably be best. Maybe a single string with namespace/name?

No need to concatenate the has function is able to construct the hash string with all the fields from the struct.

Reply:
Used Namespace, Name and UID

Reply:
I don't have clear if we should use uid, namespace/name are unique in the system, the uid is used to distinguish between objects with same namespace/name that are recreated ... can this have some undesired consequences as we create a lot of churn on the controller if there are pods that are being recreated constantly?

@liggitt WDYT?

Reply:
Maybe using namespace/name is the most conservative option for us is good enough.

Reply:
replied in https://github.com/kubernetes/kubernetes/pull/115907#issuecomment-1440340880 .
Pod addresses are eventually going to be reused, the case that a pod terminated and a pod running exist with the same IP is legit, and since we can not determine which one is the good one I think is ok to add both to the slice with the corresponding condition.

The case of two pods running with the same IP should not happen, but is a common "trick" used by networking to provide HA capabilities, so I don't think we should break it if someone wants to play with that fire ;)